tests/crypto/cert_tests.c

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/evidence.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include <string.h>
+#if defined(OE_USE_OPENSSL)
+#include "../../../enclave/crypto/openssl/cert.h"
+#endif
+
+/* Test the internal API that is only avaiable on OpenSSL-based implementation.
+ */
+#if defined(OE_USE_OPENSSL)
+static void _test_X509_parse_name()
+{
+    X509_NAME* name = NULL;
+    char buf[256];
+
+    printf("=== begin %s()\n", __FUNCTION__);
+
+    /* Basic case */
+    name = X509_parse_name("CN=Open Enclave SDK,O=OESDK TLS,C=US");
```

Comment:
I'm always leery of adding new string parsers because of the testing surface involved. Consider expanding the test cases to consider or things we need to explicitly block to scope testing. Looking at [RFC 4514](https://tools.ietf.org/html/rfc4514)
- UTF-8 strings and canonicalization of name segments 
- Special character handling
- Escape handling of things other than commas

Reply:
Current implementation is based on Mbed TLS's (https://github.com/openenclave/openenclave-mbedtls/blob/2b735b85590e661870bc8217cc09139ba17e9a6f/library/x509_create.c#L153). I believe the most of cases you described are not handled by this implementation. Would you suggest adding negative tests on them?

Reply:
Yes, if we're keeping the OE specific implementation, we need to add tests to evaluate what we think we support. As mentioned in the above comments, OpenSSL CLI will support multiRDN names (e.g. handling "+"), but MbedTLS does not appear to. The UTF-8 vs MBCS handling behavior is also unclear to me.

Reply:
Added positive tests for the UTF-8 testing and no multiRDN support, and a negative test for escaping characters other than commas.